http/1.6/modules/ROOT/pages/http-authentication.adoc

```diff
@@ -581,63 +545,50 @@ This example includes two `oauth:custom-parameter` child elements that define pa
 
-You configure the external redirect URI by setting the External Callback URL attribute (`externalCallbackUrl` in XML).
-
-Using `externalCallbackUrl` is particularly useful for deploying applications to CloudHub, for example. When configuring authentication, you can optionally specify the `localCallbackUrl` attribute as well.
+To override the redirect URI (external `redirect_uri`) configure the external redirect URI in the *External callback url* field, which is useful for deploying applications to CloudHub, for example. When you configure authentication, you can optionally configure the *Local callback url* field as well.
 
-For example, the `localCallbackUrl` is `+http://localhost:8082/callback+` in the <<create-the-mule-client-app,previous example>>:
-
-To create the endpoint for CloudHub, Mule has to create an endpoint for CloudHub in a different format. For example:
+To create the endpoint for CloudHub, Mule has to create an endpoint for CloudHub in a different format, for example:
 
 `+https://<app>.cloudhub.io/<redirect Uri>+`
 
-To instruct Mule to create the endpoint for CloudHub in the correct format, include the `externalCallbackUrl` attribute in your `oauth:authorization-code-grant-type` configuration.
-
-=== Extract Parameters from the Token URL Response
+To instruct Mule to create the endpoint for CloudHub in the correct format, include the *External callback url* field in your OAuth2 Authorization code grant type configuration.
 
-After you have obtained an authorization code from the authentication server, the OAuth dance makes a request to the Token URL of the server to receive an access token.
+=== Configure OAuth2 Authorization Extract Parameters
 
-The format of the response to the request to the token URL is not defined in the OAuth spec. Each implementation may therefore return different response formats. By default, Mule expects the response to be in JSON format. When this is the case, HTTP Request knows how to extract the required information, as long as its elements are named as below:
+After you obtained an authorization code from the authentication server, the OAuth dance makes a request to the token URL of the server to receive an access token.
 
-* *Response access token*: The JSON field is named `access_token`
-* *Response refresh token*: The JSON field is named `refresh_token`
-* *Response expires in*: The JSON field is named `expires_in`
+The format of the response to the token URL is not defined in the OAuth specification. Each implementation may return different response formats. By default, Mule expects the response to be in JSON format. When this is the case, HTTP requests knows how to extract the required information, as long as the following fields are specified:
```

Comment:
GLOBALLY review your use of "response": a "request" goes "to" something and a "response" comes "from" something (or something "responds to" something else). 
As a beginning audience, I get confused by the direction intended when this narrative uses "response". You might check this with a peer to see if they think this might confuse other audiences.

Reply:
Do you intend "may" (permission) or "might" (possibility), or "can" (ability)?

Reply:
"HTTP requests knows" > "HTTP requests know"